Import host-server-manager and rename it to host-runner

gligneul opened the pull request:
Close #23

gligneul committed:
refactor(offchain): import host-server-manager

The original history can be found at the previous repository.
https://github.com/cartesi/host-server-manager

gligneul committed:
refactor(offchain): transform host-server-manager into host-runner

- Rename the host-server-manager to host-runner.
- Remove redundant files from the host-runner crate.
- Remove dates from copyright.
- Replace env-logger with tracing.
- Replace structopt with clap.
- Use grpc-interfaces crate in host-runner.
- Add host-runner to the offchain workspace Cargo.toml.
- Add host-runner to the offchain Dockerfile.
- Add host-runner to the Github CI.

renan061 commented on offchain/host-runner/src/config.rs:
```diff
@@ -0,0 +1,51 @@
+// CONDITIONS OF ANY KIND, either express or implied. See the License for the
+// specific language governing permissions and limitations under the License.
+
+use structopt::StructOpt;
```
We are still using `structopt` here instead of `clap`. Do we have an issue for that?

gligneul replied:
I will change it to clap now.

gligneul commented on offchain/host-runner/LICENSE:
```diff
@@ -0,0 +1,201 @@
+                                 Apache License
```
I will remove this file.

gligneul committed:
refactor(offchain): import rollup-http-client

The original code history is available at the machine emulator tools
repository (https://github.com/cartesi/machine-emulator-tools/).

gligneul committed:
refactor(offchain): adapt rollup-http-client to rollup workspace

torives reviewed (commented):
Great job! Thanks for the sensible commits, made the review a lot easier. Eager to see the future improvements to the host-runner once this gets merged! 

Just found some minor details and typos. Also, there's a "r" missing in the branch's name 😅

gligneul committed:
test(host-runner): fix race condition in tests

The race condition happened because the finish function in the
finish_advance_state function did not wait for the result.
